### PR TITLE
Div 4136 use govuk frontend 

### PR DESCRIPTION
**Pull request by dtaylor7**
[DIV-3893 - Update to GOV.UK Frontend](https://tools.hmcts.net/jira/browse/DIV-3893)
 - Remove govuk_template
 - Add govuk-frontend
 - Update html to work with new gov.uk frontend

### CODE REVIEW

File: app/assets/javascripts/documentUpload.js [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
   var documentUpload = function() {
 
+    console.log('here');
```
Comment (yarpo) [inline app/assets/javascripts/documentUpload.js]: is this what we want to merge into master?

File: app/components/AddressLookupStep/template.html [hunk 2]
```diff
@@ -3,158 +3,150 @@
...
-        <div class="form-group">
-          <input type="hidden" name="addressType" value="manual"/>
-          <input type="hidden" name="addressConfirmed" value="true"/>
+      <fieldset class="govuk-fieldset">
+        <legend class="govuk-fieldset__legend govuk-visually-hidden">{{content.fullAddress}}</legend>
+        <div class="govuk-form-group {{ 'govuk-form-group--error' if fields.addressManual.error }}">
+          <label class="govuk-label" for="addressManual">{{content.fullAddress}}</label>
+
+          {{ errorsFor(fields.addressManual) }}
+          <textarea class="govuk-textarea"
```
Comment (yarpo) [inline app/components/AddressLookupStep/template.html]: why not macro?

File: app/steps/grounds-for-divorce/reason/partials/checkYourAnswersTemplate.html [hunk 3]
```diff
@@ -1,6 +1,6 @@
-<tr>
-  <th>{{ content.question | safe }}</th>
-  <td>
+<tr class="govuk-table__row">
+  <td class="govuk-table__cell">{{ content.question | safe }}</th>
```
Comment (yarpo) [inline app/steps/grounds-for-divorce/reason/partials/checkYourAnswersTemplate.html]: `<td>` and  `</th>`?

File: app/steps/help/need-help/partials/checkYourAnswersTemplate.html [hunk 4]
```diff
@@ -1,5 +1,5 @@
-<tr>
-  <th>{{ content.question | safe }}</th>
-  <td>{{ fields.helpWithFeesNeedHelp.value }}</td>
-  <td><a class="link" aria-label="{{ content.change }} {{ content.question | safe }}" href="{{ content.url | safe }}">{{ content.change }}</a></td>
+<tr class="govuk-table__row">
+  <td class="govuk-table__cell">{{ content.question | safe }}</th>
```
Comment (yarpo) [inline app/steps/help/need-help/partials/checkYourAnswersTemplate.html]: `<td>` and `</th>`?

File: app/assets/javascripts/documentUpload.js (hunk 1 above)
Comment (dtaylor7) [reply app/assets/javascripts/documentUpload.js]: absolutely not 👍

File: app/steps/jurisdiction/domicile/partials/checkYourAnswersTemplate.html [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
+  <td class="govuk-table__cell">{{ content.petitioner | safe }}</th>
+  <td class="govuk-table__cell">{{ fields.jurisdictionPetitionerDomicile.value }}</td>
+  <td class="govuk-table__cell"><a class="govuk-link" aria-label="{{ content.change }} {{ content.petitioner | safe }}" href="{{ content.url | safe }}">{{ content.change }}</a></td>
 </tr>
-<tr>
-  <th>{{ content.respondent | safe }}</th>
-  <td>{{ fields.jurisdictionRespondentDomicile.value }}</td>
-  <td><a class="link" aria-label="{{ content.change }} {{ content.respondent | safe }}" href="{{ content.url | safe }}">{{ content.change }}</a></td>
+<tr class="govuk-table__row">
+  <td class="govuk-table__cell">{{ content.respondent | safe }}</th>
```
Comment (yarpo) [inline app/steps/jurisdiction/domicile/partials/checkYourAnswersTemplate.html]: td th

File: app/steps/jurisdiction/domicile/template.html [hunk 6]
```diff
@@ -7,66 +7,75 @@
...
-        name = 'jurisdictionPetitionerDomicile',
-        field = fields.jurisdictionPetitionerDomicile,
-        dataTarget = 'jurisdictionRespondentDomicile',
-        screenReader = content.petitioner
-        ) }}
+            {{ errorsFor(fields.jurisdictionPetitionerDomicile) }}
 
-    </fieldset>
+            <div class="govuk-radios govuk-radios--inline">
+                {{ radioButton(
```
Comment (yarpo) [inline app/steps/jurisdiction/domicile/template.html]: can you please format this fragment a little bit better?

File: app/steps/jurisdiction/domicile/template.html [hunk 7]
```diff
@@ -7,66 +7,75 @@
...
-        screenReader = content.respondent
-        ) }}
+    <div id="jurisdictionRespondentDomicile" class="govuk-form-group {{ 'js-hidden' if fields.jurisdictionPetitionerDomicile.value == undefined }} {{ 'govuk-form-group--error' if fields.jurisdictionRespondentDomicile.error }}">
+        <fieldset class="govuk-fieldset">
+            <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">{{ content.respondent | safe }}</legend>
 
-    </fieldset>
+            {{ errorsFor(fields.jurisdictionRespondentDomicile) }}
+            <div class="govuk-radios govuk-radios--inline">
+                {{ radioButton(
```
Comment (yarpo) [inline app/steps/jurisdiction/domicile/template.html]: the same

File: app/steps/jurisdiction/habitual-residence/partials/checkYourAnswersTemplate.html [hunk 8]
```diff
@@ -1,10 +1,10 @@
...
+  <td class="govuk-table__cell">{{ content.petitioner | safe }}</th>
+  <td class="govuk-table__cell">{{ fields.jurisdictionPetitionerResidence.value }}</td>
+  <td class="govuk-table__cell"><a class="govuk-link" aria-label="{{ content.change }} {{ content.petitioner | safe }}" href="{{ content.url | safe }}">{{ content.change }}</a></td>
 </tr>
-<tr>
-  <th>{{ content.respondent | safe }}</th>
-  <td>{{ fields.jurisdictionRespondentResidence.value }}</td>
-  <td><a class="link" aria-label="{{ content.change }} {{ content.respondent | safe }}" href="{{ content.url | safe }}">{{ content.change }}</a></td>
+<tr class="govuk-table__row">
+  <td class="govuk-table__cell">{{ content.respondent | safe }}</th>
```
Comment (yarpo) [inline app/steps/jurisdiction/habitual-residence/partials/checkYourAnswersTemplate.html]: td th

File: app/steps/jurisdiction/habitual-residence/template.html [hunk 9]
```diff
@@ -7,55 +7,66 @@
...
+
+        <div class="govuk-form-group {{ 'govuk-form-group--error' if fields.jurisdictionPetitionerResidence.error }}">
+            <fieldset id="jurisdictionPetitionerResidence" class="govuk-fieldset">
+                <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">{{ content.petitioner | safe }}</legend>
+                <p class="govuk-body">{{ content.info | safe }}</p>
+
+                {{ errorsFor(fields.jurisdictionPetitionerResidence) }}
+
+                <div class="govuk-radios govuk-radios--inline">
+                    {{ radioButton(
```
Comment (yarpo) [inline app/steps/jurisdiction/habitual-residence/template.html]: reformat please

File: app/steps/jurisdiction/habitual-residence/template.html [hunk 10]
```diff
@@ -7,55 +7,66 @@
...
+        </div>
+
+        <div class="govuk-form-group">
+            <fieldset id="jurisdictionRespondentResidence" class="govuk-fieldset {{ 'js-hidden' if fields.jurisdictionPetitionerResidence.value === undefined }} {{ 'govuk-form-group--error' if fields.jurisdictionRespondentResidence.error }}">
+                <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">{{ content.respondent | safe }}</legend>
+                <p class="govuk-body">{{ content.info | safe }}</p>
+
+                {{ errorsFor(fields.jurisdictionRespondentResidence) }}
+                <div class="govuk-radios govuk-radios--inline">
+                    {{ radioButton(
```
Comment (yarpo) [inline app/steps/jurisdiction/habitual-residence/template.html]: reformat please

File: app/steps/jurisdiction/interstitial/template.html [hunk 11]
```diff
@@ -7,67 +7,69 @@
...
-            value = 'No',
-            label = content.needInfo,
-            name = 'jurisdictionConfidentLegal',
-            field = fields.jurisdictionConfidentLegal
-            ) }}
-
-        </fieldset>
+                {{ errorsFor(fields.jurisdictionConfidentLegal) }}
+                <div class="govuk-radios govuk-radios--inline">
+                    {{ radioButton(
```
Comment (yarpo) [inline app/steps/jurisdiction/interstitial/template.html]: reformat please

File: app/steps/jurisdiction/last-resort/template.html [hunk 12]
```diff
@@ -7,74 +7,81 @@
...
-        ) }}
-
-    </fieldset>
+    <div class="govuk-form-group">
+        <fieldset id="reason" class="govuk-fieldset jurisdiction{{ ' error' if fields.reason.error }}">
+            <legend class="govuk-fieldset__legend govuk-!-font-weight-bold">{{ content.select | safe }}</legend>
+
+            <div class="govuk-checkboxes">
+
+                {{ checkBoxCustom(
```
Comment (yarpo) [inline app/steps/jurisdiction/last-resort/template.html]: reformat please

File: app/steps/jurisdiction/last-six-months/template.html [hunk 13]
```diff
@@ -6,23 +6,27 @@
...
-    field = fields.jurisdictionLastSixMonths,
-    screenReader = content.question
-    ) }}
-</fieldset>
+<div class="govuk-form-group {{ 'govuk-form-group--error' if fields.jurisdictionLastSixMonths.error }}">
+    <fieldset class="govuk-fieldset inline" id="jurisdictionLastSixMonths">
+        <legend class="govuk-fieldset__legend visually-hidden">{{ content.question | safe }}</legend>
+        {{ errorsFor(fields.jurisdictionLastSixMonths) }}
+        <div class="govuk-radios govuk-radios--inline">
+            {{ radioButton(
```
Comment (yarpo) [inline app/steps/jurisdiction/last-six-months/template.html]: reformat please

File: app/steps/living-arrangements/last-lived-together/at-petitioner-home-address/partials/checkYourAnswersTemplate.html [hunk 14]
```diff
@@ -1,12 +1,12 @@
-<tr>
-  <th>
+<tr class="govuk-table__row">
+  <td class="govuk-table__cell">
```
Comment (yarpo) [inline app/steps/living-arrangements/last-lived-together/at-petitioner-home-address/partials/checkYourAnswersTemplate.html]: td th?

File: app/steps/living-arrangements/live-together/partials/checkYourAnswersTemplate.html [hunk 15]
```diff
@@ -1,5 +1,5 @@
-<tr>
-  <th>{{ content.question | safe }}</th>
-  <td>{{ fields.livingArrangementsLiveTogether.value | safe }}</td>
-  <td><a class="link" aria-label="{{ content.change }} {{ content.question | safe }}" href="{{ content.url | safe }}">{{ content.change }}</a></td>
+<tr class="govuk-table__row">
+  <td class="govuk-table__cell">{{ content.question | safe }}</th>
```
Comment (yarpo) [inline app/steps/living-arrangements/live-together/partials/checkYourAnswersTemplate.html]: td th

File: app/steps/marriage/about-your-marriage-certificate/template.html [hunk 16]
```diff
@@ -5,42 +5,38 @@
...
-            name ='certificateInEnglish',
-            field = fields.certificateInEnglish,
-            yes = content.yes,
-            no = content.no,
-            targetYes = '',
-            targetNo = 'certified-translation',
-            screenReader = content.question
-            ) }}
-        </fieldset>
+    {{ yesNoRadioHeading(
```
Comment (yarpo) [inline app/steps/marriage/about-your-marriage-certificate/template.html]: reformat please

File: app/steps/marriage/about-your-marriage-certificate/template.html [hunk 17]
```diff
@@ -5,42 +5,38 @@
...
-            name ='certifiedTranslation',
-            field = fields.certifiedTranslation,
-            yes = content.yes,
-            no = content.no,
-            screenReader = content.certifiedTranslation
-            ) }}
-        </fieldset>
+    <p class="govuk-body">{{ content.certifiedDescription | safe }}</p>
+
+    {{ yesNoRadioHeading(
```
Comment (yarpo) [inline app/steps/marriage/about-your-marriage-certificate/template.html]: reformat please

File: app/steps/marriage/certificate-names/partials/checkYourAnswersTemplate.html [hunk 18]
```diff
@@ -1,10 +1,10 @@
-<tr>
-  <th>{{ content.marriagePetitionerName | safe }}</th>
-  <td>{{ fields.marriagePetitionerName.value }}</td>
-  <td><a class="link" aria-label="{{ content.change }} {{ content.marriagePetitionerName | safe }}" href="{{ content.url | safe }}">{{ content.change }}</a></td>
+<tr class="govuk-table__row">
+  <td class="govuk-table__cell">{{ content.marriagePetitionerName | safe }}</th>
```
Comment (yarpo) [inline app/steps/marriage/certificate-names/partials/checkYourAnswersTemplate.html]: td th

File: app/steps/marriage/certificate-names/template.html [hunk 19]
```diff
@@ -12,69 +13,67 @@
...
+        {{ content.questionTrans | safe}}
+      {%- else -%}
+        {{ content.question | safe}}
+      {%- endif -%}
+    </legend>
 
+  {%- if session.certificateInEnglish == 'No' -%}
 
-{%- if session.certificateInEnglish == 'No' -%}
+    {{ textField(
```
Comment (yarpo) [inline app/steps/marriage/certificate-names/template.html]: reformat please

File: app/steps/marriage/date/partials/checkYourAnswersTemplate.html [hunk 20]
```diff
@@ -1,5 +1,5 @@
-<tr>
-  <th>{{ content.question | safe }}</th>
-  <td>{{ fields.marriageDate.value }}</td>
-  <td><a class="link" aria-label="{{ content.change }} {{ content.question | safe }}" href="{{ content.url | safe }}">{{ content.change }}</a></td>
+<tr class="govuk-table__row">
+  <td class="govuk-table__cell">{{ content.question | safe }}</th>
```
Comment (yarpo) [inline app/steps/marriage/date/partials/checkYourAnswersTemplate.html]: td th

File: app/steps/marriage/foreign-certificate/partials/checkYourAnswersTemplate.html [hunk 21]
```diff
@@ -1,19 +1,19 @@
-<tr>
-  <th>{{ content.countryName | safe }}</th>
-  <td>{{ fields.countryName.value }}</td>
-  <td><a class="link" aria-label="{{ content.change }} {{ content.countryName | safe }}" href="{{ content.url | safe }}">{{ content.change }}</a></td>
+<tr class="govuk-table__row">
+  <td class="govuk-table__cell">{{ content.countryName | safe }}</th>
```
Comment (yarpo) [inline app/steps/marriage/foreign-certificate/partials/checkYourAnswersTemplate.html]: td th

File: app/components/AddressLookupStep/template.html (hunk 2 above)
Comment (dtaylor7) [reply app/components/AddressLookupStep/template.html]: There are macro's for many of the changes i have made here. However it was far less time consuming to do global search and replace on classes compared with changing html to macro's ( even with that it has taken weeks ).

File: app/steps/grounds-for-divorce/reason/partials/checkYourAnswersTemplate.html (hunk 3 above)
Comment (dtaylor7) [reply app/steps/grounds-for-divorce/reason/partials/checkYourAnswersTemplate.html]: 👍  resolved this on many pages. good catch

File: app/steps/jurisdiction/domicile/template.html (hunk 6 above)
Comment (dtaylor7) [reply app/steps/jurisdiction/domicile/template.html]: errrm... no?

File: app/steps/marriage/certificate-names/template.html (hunk 19 above)
Comment (dtaylor7) [reply app/steps/marriage/certificate-names/template.html]: im not refactoring all code here... just minimum updates and changes to get to new govuk design system... please dont make the job any harder and more time consuming than it already is....

File: app/steps/living-arrangements/live-together/partials/checkYourAnswersTemplate.html (hunk 15 above)
Comment (dtaylor7) [reply app/steps/living-arrangements/live-together/partials/checkYourAnswersTemplate.html]: 👍

File: app/components/AddressLookupStep/template.html [hunk 22]
```diff
@@ -3,158 +3,150 @@
...
-    {% block warning %}{% endblock %}
+  {% block warning %}{% endblock %}
 
-    {% if errors.length > 0 %}
-      <div class="error-summary" role="group" aria-labelledby="error-summary-heading" tabindex="-1">
-
-        <h1 class="heading-medium error-summary-heading" id="error-summary-heading">
-          {{common.errorSummaryHeading | safe}}
+  {% if errors.length > 0 %}
+      <div class="govuk-error-summary" aria-labelledby="error-summary-title" role="alert" tabindex="-1" data-module="error-summary">
```
Comment (iichr) [inline app/components/AddressLookupStep/template.html]: According to https://design-system.service.gov.uk/components/error-summary/, the error summary needs to precede the heading:
> Put the error summary at the top of the main container. If your page includes breadcrumbs or a back link, place it below these, but above the `<h1>`.

File: app/steps/check-your-answers/template.html [hunk 23]
```diff
@@ -1,146 +1,141 @@
...
+  <h1 class="govuk-heading-l">
+      {%- if fields.nextStepUrl.value -%}
+        {{ content.titleSoFar }}
+      {%- else -%}
+        {{ content.title }}
+      {%- endif -%}
+  </h1>
+
+  {% if errors.length > 0 and not fields.nextStepUrl.value %}
+    <div class="govuk-error-summary" aria-labelledby="error-summary-title" role="alert" tabindex="-1" data-module="error-summary">
```
Comment (iichr) [inline app/steps/check-your-answers/template.html]: same here https://design-system.service.gov.uk/components/error-summary/

File: app/components/AddressLookupStep/template.html (hunk 2 above)
Comment (yarpo) [reply app/components/AddressLookupStep/template.html]: ok

File: app/steps/jurisdiction/domicile/partials/checkYourAnswersTemplate.html [hunk 24]
```diff
@@ -1,10 +1,10 @@
-<tr>
-  <th>{{ content.petitioner | safe }}</th>
-  <td>{{ fields.jurisdictionPetitionerDomicile.value }}</td>
-  <td><a class="link" aria-label="{{ content.change }} {{ content.petitioner | safe }}" href="{{ content.url | safe }}">{{ content.change }}</a></td>
+<tr class="govuk-table__row">
+  <td class="govuk-table__cell">{{ content.petitioner | safe }}</th>
```
Comment (yarpo) [inline app/steps/jurisdiction/domicile/partials/checkYourAnswersTemplate.html]: `<td>`, but then `</th>` - is that correct?

File: app/steps/jurisdiction/domicile/template.html (hunk 6 above)
Comment (yarpo) [reply app/steps/jurisdiction/domicile/template.html]: why not?

**Comment by dtaylor7** [comment]
close until ready to merge

File: app/steps/jurisdiction/domicile/template.html (hunk 6 above)
Comment (dtaylor7) [reply app/steps/jurisdiction/domicile/template.html]: just working on upgrading to govuk design kit which is enough work. we can reformat in another tech debt

File: app/components/AddressLookupStep/template.html (hunk 22 above)
Comment (dtaylor7) [reply app/components/AddressLookupStep/template.html]: yep we can change that if required, but i don't want to move elements unless 100% necessary without design and approval from business. therefor am going to leave this one

**Pull request closed without merging**